packages/web/src/pages/oauth-login-page/hooks.ts

```diff
@@ -25,7 +25,14 @@ export const useParsedQueryParams = () => {
         return 'postmessage'
       }
       try {
-        return new URL(decodeURIComponent(redirectUri))
+        const url = new URL(decodeURIComponent(redirectUri))
+        // Ensure that the redirect_uri protocol is https
+        // IMPORTANT: If this validation is not done, users can
+        // use the redirect_uri to execute arbitrary code on the host
+        // domain (e.g. audius.co).
+        if (url.protocol !== 'https:') {
```

Comment:
@raymondjacobson can you move this to `getIsRedirectValid`?

Reply:
nice, yup will do